### PR TITLE
CXF 3.X and jaxws-2.2, wsSecurity-1.1, wsAtomicTransation-1.2 feature, runtime, and test updates 

### PR DESCRIPTION
**Pull request by arunavemulapalli**
This is required to support jaxws-2.2 feature update

### CODE REVIEW

**Comment by neuwerk** [comment]
https://wasrtc.hursley.ibm.com:9443/jazz/web/projects/WS-CD#action=com.ibm.team.build.viewResult&id=_2OYbsPIqEeyYXLZn0n3zUg&tab=com.ibm.team.build.web.ui.internal.editors.result.test.TestProvider

File: dev/com.ibm.ws.org.apache.wss4j.ws.security.common.2.3.0/src/org/apache/ws/security/WSPasswordCallback.java [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+package org.apache.ws.security;
+
+import java.security.Key;
+
+import javax.security.auth.callback.Callback;
+
+import org.w3c.dom.Element;
+
+/**
+ *
```
Comment (neuwerk) [inline dev/com.ibm.ws.org.apache.wss4j.ws.security.common.2.3.0/src/org/apache/ws/security/WSPasswordCallback.java:20]: For code maintainability I think we should add a comment about why this has been added and what it does.

File: dev/com.ibm.ws.org.apache.wss4j.ws.security.common.2.3.0/src/org/apache/ws/security/WSSecurityException.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.ws.security;
+
+
+
+/**
+ * Exception class for WS-Security.
```
Comment (neuwerk) [inline dev/com.ibm.ws.org.apache.wss4j.ws.security.common.2.3.0/src/org/apache/ws/security/WSSecurityException.java:25]: Same as the other class, I think we should add a short comment about why we added this class

File: dev/com.ibm.ws.org.apache.wss4j.ws.security.common.2.3.0/src/org/apache/ws/security/components/crypto/Crypto.java [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+import java.security.PublicKey;
+import java.security.cert.CertificateFactory;
+import java.security.cert.X509Certificate;
+import java.util.Collection;
+import java.util.regex.Pattern;
+
+import javax.security.auth.callback.CallbackHandler;
+//import org.apache.ws.security.WSSecurityException;
+
+
```
Comment (neuwerk) [inline dev/com.ibm.ws.org.apache.wss4j.ws.security.common.2.3.0/src/org/apache/ws/security/components/crypto/Crypto.java:32]: Same as the other class, I think we should add a short comment about why we added this class

File: dev/com.ibm.ws.org.apache.wss4j.ws.security.common.2.3.0/src/org/apache/ws/security/components/crypto/CryptoFactory.java [hunk 4]
```diff
@@ -0,0 +1,237 @@
...
+import java.io.InputStream;
+import java.lang.reflect.Constructor;
+import java.util.Map;
+import java.util.Properties;
+
+import org.apache.wss4j.common.ext.WSSecurityException;
+import org.apache.wss4j.common.util.Loader;
+
+/**
+ * CryptoFactory.
```
Comment (neuwerk) [inline dev/com.ibm.ws.org.apache.wss4j.ws.security.common.2.3.0/src/org/apache/ws/security/components/crypto/CryptoFactory.java:32]: Same as the other class, I think we should add a short comment about why we added this class

File: dev/com.ibm.ws.org.apache.wss4j.ws.security.common.2.3.0/src/org/apache/wss4j/common/crypto/Crypto.java [hunk 5]
```diff
@@ -0,0 +1,228 @@
...
+import java.security.PublicKey;
+import java.security.cert.CertificateFactory;
+import java.security.cert.X509Certificate;
+import java.util.Collection;
+import java.util.regex.Pattern;
+
+import javax.security.auth.callback.CallbackHandler;
+
+import org.apache.wss4j.common.ext.WSSecurityException;
+
```
Comment (neuwerk) [inline dev/com.ibm.ws.org.apache.wss4j.ws.security.common.2.3.0/src/org/apache/wss4j/common/crypto/Crypto.java:32]: Same as the other class, I think we should add a short comment about why we added this class

File: dev/com.ibm.ws.org.apache.wss4j.ws.security.common.2.3.0/src/org/apache/wss4j/common/crypto/CryptoFactory.java [hunk 6]
```diff
@@ -0,0 +1,294 @@
...
+import java.io.InputStream;
+import java.lang.reflect.Constructor;
+import java.util.Map;
+import java.util.Properties;
+
+import org.apache.wss4j.common.ext.WSSecurityException;
+import org.apache.wss4j.common.util.Loader;
+
+/**
+ * CryptoFactory.
```
Comment (neuwerk) [inline dev/com.ibm.ws.org.apache.wss4j.ws.security.common.2.3.0/src/org/apache/wss4j/common/crypto/CryptoFactory.java:32]: Same as the other class, I think we should add a short comment about why we added this class

Comment (neuwerk) [reply dev/com.ibm.ws.org.apache.wss4j.ws.security.common.2.3.0/src/org/apache/wss4j/common/crypto/CryptoFactory.java:32]: Are there Liberty specific changes in this class? Just curious as I don't see them, so if not do we need to be recompiling the `org.apache.wss4j.common.crypto` and `org.apache.wss4j.common.ext` APIs?

File: dev/com.ibm.ws.org.apache.wss4j.ws.security.common.2.3.0/src/org/apache/wss4j/common/ext/WSSecurityException.java [hunk 7]
```diff
@@ -26,7 +26,7 @@
 /**
  * Exception class for WS-Security.
  */
-//No Liberty code change
+// Liberty code change
```
Comment (neuwerk) [inline dev/com.ibm.ws.org.apache.wss4j.ws.security.common.2.3.0/src/org/apache/wss4j/common/ext/WSSecurityException.java:29]: What's the Liberty change here? Just curious if it's denoted?

**Review by sawadood — COMMENTED** [review]
The dynamic trace updated that was done for BPM when using Logging interceptor classes in trace string, will the BPM update to support dynamic trace still work without any updates to the application or configuration?

**Review by sawadood — COMMENTED** [review]
The dynamic trace updated that was done for BPM when using Logging interceptor classes in trace string, will the BPM update to support dynamic trace still work without any updates to the application or configuration?

In com.ibm.websphere.appserver.wsSecurity-1.1.feature, why are some packages have type as "third-party"?  I don't think any packages in this feature are shipped as third-party APIs

**Review by neuwerk — COMMENTED** [review]
I have some questions about the changes to the test buckets, but I will follow up with you offline.

**Comment by wraschke** [comment]
#libby

**Comment by neuwerk** [comment]
Closing as this was merged by #21587

**Pull request closed without merging**